### PR TITLE
Map visualisations - geo variables/entity chooser via geoConfig + per-study disabling of new-viz-icons infrastructure

### LINKED ISSUE #776: Map visualization: enable geolocation selection in `<InputVariables>`
This follows #775 

The map visualization should have a "Geolocation" axis variable picker.  This should allow the user to pick a geolocation (EUPATH_0043212) node in the variable tree if that node has lat, long, and geohash variables below it.  Note that those vars are always hidden.

This may require work upstream in the variable annotation department (and possibly introduce a new constraint), or we could inspect the variable tree and figure out which entities have suitable geolocations.

Ultimately, this is an entity picker, not a variable picker, because we are limited to having only one pair of latitude and longitude variables per entity - unless we make multiple flavours of variable, e.g., `home address latitude`, `work address latitude` etc.

### LINKED ISSUE #796: Map: gray out the map viz tool in visualizations when the study has no geolocation variables
Not sure how to do this yet!

We may have a better idea after #776

### PR DESCRIPTION
**Pull request by bobular**
Resolves #796 
Resolves #776 

This PR has two prongs.

1. develops the map viz further by adding two dropdown menus to select the mappable entity and the display count entity (wording for these needs to be refined).

2. adds some hopefully general infrastructure for enabling/disabling visualisations from the "new viz picker" when there is no chance that they can be used in a study. 

**Demo/review only with UMSP Study!**
(Although you can check out the disabled map viz icon in other studies.)

Very basic UI added to allow 'geo entity' and 'output/count entity' to be chosen.
The latter cannot be a parent entity of the former (need to double check this is correct).

Uses Material UI `<Select>` but ideally we could have our own core-ui component.

Look and feel is inconsistent with variable tree pickers but for now it's OK. We'll await further work in core-components and revisit.

### CODE REVIEW

File: src/lib/core/components/visualizations/VisualizationsContainer.tsx [hunk 1]
```diff
@@ -230,15 +238,22 @@ function NewVisualizationPicker(props: Props) {
...
               <div className={cx('-PickerEntryName')}>
-                {vizOverview.displayName?.includes(', ') ? (
-                  <div>
-                    {vizOverview.displayName.split(', ')[0]} <br />
-                    {vizOverview.displayName.split(', ')[1]}
-                  </div>
-                ) : (
-                  <div>{vizOverview.displayName}</div>
-                )}
+                <div>
```
Comment (bobular) [inline src/lib/core/components/visualizations/VisualizationsContainer.tsx:241]: 8 lines become 12!

File: src/lib/core/hooks/geoConfig.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * Assumes that aggregationVariables are in ID order (e.g. geohash_1 has the lowest ID and geohash_6 the highest)
+ *
+ */
+
+export function useGeoConfig(entities: StudyEntity[]): GeoConfig[] {
+  return useMemo(
+    () =>
+      entities
+        .map((entity) => {
+          const geolocationNode = findGeolocationNode(entity);
```
Comment (bobular) [inline src/lib/core/hooks/geoConfig.ts:21]: A lot of the logic in `findGeolocationNode` is repeated below.

Is there a better way to "filter and map" at the same time?

**Comment by chowington** [comment]
@bobular I'm getting an issue in testing: When I choose a counts entity that's not Households, minimize the viz, then go back into the viz, the markers don't load

**Review by chowington — APPROVED** [review]
One last thing: be sure to check the es-lint warnings! I think these changes add a couple of warnings about unused variables. If you're planning on using these variables later, no prob 🙂

File: src/lib/core/components/visualizations/VisualizationsContainer.tsx [hunk 3]
```diff
@@ -230,15 +238,22 @@ function NewVisualizationPicker(props: Props) {
...
+                  {vizOverview.displayName
+                    ?.split(', ') // this monstrosity avoids doing the split twice!
+                    .reduce<(string | JSX.Element)[]>(
+                      (accum, item) => [
+                        ...accum,
+                        ...(accum.length ? [<br />] : []),
+                        item,
+                      ],
+                      []
+                    )}
```
[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/components/visualizations/VisualizationsContainer.tsx:251]: Looks much better! Did you try that in anger? If not, I will give it a try.

I did have shorter versions but they didn't work!

File: src/lib/core/utils/geoVariables.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      (variable) => variable.type === 'number'
+    );
+    if (numberSiblings.length === 1) {
+      const stringCategoricals = siblingVariables.filter(
+        (variable) =>
+          variable.type === 'string' &&
+          (variable.dataShape === 'categorical' ||
+            variable.dataShape === 'binary')
+        // two lines above, it wouldn't let me do ({ type, dataShape}) => ...
+        // without jumping through more typing hoops - any idea why please?
```
[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/utils/geoVariables.ts:37]: Aha - sweet explanation - thanks!

File: src/lib/core/components/visualizations/VisualizationsContainer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/components/visualizations/VisualizationsContainer.tsx:251]: It works and there are no TS errors in my IDE :-) 

I think it's a lot more readable. Very nice, thanks!

**Comment by bobular** [comment]
> Not in the sense you're meaning; I'd normally filter first, then map, but in this case it wouldn't be any more performant. To cut down on duplication, I wonder if you could combine this logic and the findGeolocationNode logic into a function that both finds the node and transforms it at the same time? Unless you plan on using findGeolocationNode elsewhere, in which case I'm not seeing a great way to DRY this up.

Thanks @chowington - I don't think I do plan on using findGeoocationNode anywhere else so I followed your suggestion and factored out the repeated code. :-)

**Pull request merged**